Formula/unyaffs.rb

```diff
@@ -1,9 +1,17 @@
 class Unyaffs < Formula
   desc "Extract files from a YAFFS2 filesystem image"
   homepage "https://git.b-ehlers.de/ehlers/unyaffs/"
-  url "https://git.b-ehlers.de/ehlers/unyaffs/archive/0.9.7.tar.gz"
-  sha256 "17489fb07051d228ede6ed35c9138e25f81085492804104a8f52c51a1bd6750d"
-  head "https://git.b-ehlers.de/ehlers/unyaffs.git"
+  # The tarballs in the /archive/ directory seem to be auto-generated
```

Comment:
did we create an upstream issue for this? It sounds like their release process issue.

Reply:
I emailed him this morning when I first hit the sha256 issue.  No response yet, but it is the holidays.

If you look at the `last-modified` header from that `0.9.7.tar.gz` URL you'll see a date from today (although it does seem like it caches it a bit... I got the same value the couple times I tried)  Very suggestive of a dynamically generated artifact.

Reply:
> The tarballs seem to be generated by gitea on the fly, so the sha256 does not remain constant long-term.
> 
> I have emailed the maintainer to verify this theory, but it seems that the "last-modified" http header for https://git.b-ehlers.de/ehlers/unyaffs/archive/0.9.7.tar.gz is always recent which points towards a dynamically-generated response (albeit probably with a bit of caching)

I'm using gitea on <https://git.b-ehlers.de/>, which generates the archive by running the `git archive` command. So yes, the archive is created on the fly and then cached for some time. I'm not sure, if that can lead to a the changed sha256.

I verified the file contents with that from Debian (unyaffs_0.9.7.orig.tar.gz downloaded from <https://packages.debian.org/buster/unyaffs>), its content hasn't changed. So I'm unsure, what caused the change of the sha256 checksum.

Reply:
> unyaffs_0.9.7.orig.tar.gz downloaded from https://packages.debian.org/buster/unyaffs), its content hasn't changed.

```
$ sha256sum unyaffs_0.9.7.orig.tar.gz
099ee9e51046b83fe8555d7a6284f6fe4fbae96be91404f770443d8129bd8775  unyaffs_0.9.7.orig.tar.gz
```

The version of the `archive/0.9.7.tar.gz` file from gitea when this formula was updated a year ago was 17489fb07051d228ede6ed35c9138e25f81085492804104a8f52c51a1bd6750d

Yesterday when I fetched it it had sha256 792d18c3866910e25026aaa9dcfdec4b67bca7453ce5b2474d1ce8e9d31b2c69 

That "792d..." one is still the value being returned by git.b-ehlers.de but I'm guessing that just means that it is still cached.

So I guess one option would be to just switch the formula to use that 79d2 sha256 and hope it doesn't change again.  Or do what I am here and just pull the source from git directly.

Reply:
I just removed the cached `archive/0.9.7.tar.gz` within gitea and re-downloaded it, so it got newly build. It still has the new sha256 `792d18c3866910e25026aaa9dcfdec4b67bca7453ce5b2474d1ce8e9d31b2c69`. so currently it seems to be stable.

The debian `unyaffs_0.9.7.orig.tar.gz` has yet another sha256 as debian has packed the source archive their own way. What I did was a file compare of each file in the debian source with that of my source. And that comparison was OK.